src/main/java/com/example/springboot/controller/HelloController.java

```diff
@@ -1,24 +1,48 @@
 package com.example.springboot.controller;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.example.springboot.ShoppingListService;
+import com.example.springboot.service.Product;
 
 @RestController
+@RequestMapping("/api/v1/item")
 public class HelloController {
+	List<Product> ls = new ArrayList<Product>();
+	
 	@Autowired
 	ShoppingListService sls;
 	
-	@GetMapping("/getShop")
-	public int getShop(int n) {
-		return sls.incrementValue(n);
+	@DeleteMapping
+	public int removeProduct(@RequestParam("name")String name) {
+		for(Product pr:ls) {
+			if(pr.getName()==name) {
+				ls.remove(pr);
+				return 200;
+			}
+		}
+		return 404;
 	}
-	
-	@GetMapping("/")
-	public String index() {
-		return "Greetings from Spring Boot!";
+	@GetMapping
+	public List<Product> showList() {
+		return ls;
+	}
+	@PostMapping
+	public int addProduct(Product p) {
```

Comment:
Don't name variables or parameters with one letter. Always use a full functionally descriptive name (`product` may be enough here). The same for the list `ls`